[kanna-1] iP

kanna-1 opened the pull request:
# Arona
> "Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Arona frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~  *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/110)
2. double-click it
3. add your tasks
4. let it manage tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing Tasks
- [ ] Managing Deadline (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java Programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

kanna-1 committed:
Renamed Duke, added Greet and Exit.

kanna-1 committed:
Added Echo functionality.

kanna-1 committed:
Added Add and List functionality. Made inputs case insensitive.

kanna-1 committed:
Created a Task class. Added Mark as Done functionality.

kanna-1 committed:
Added Deadline, Event and Todo Tasks. Changed personality of the chatbot.

kanna-1 committed:
Implemented Automated Testing using Input/Output Redirection.

kanna-1 committed:
Handled errors due to invalid input by user.

kanna-1 committed:
Added delete functionality.

kanna-1 committed:
Add save and load functionality

kanna-1 committed:
Merge branch 'branch-Level-7'

kanna-1 committed:
Store deadline dates as a LocalDate in task objects

kanna-1 committed:
Merge branch 'branch-Level-8'

kanna-1 committed:
Use more OOP

kanna-1 committed:
Organize into Packages

kanna-1 committed:
Change project structure and naming

kanna-1 committed:
Merge branch 'add-gradle-support'

kanna-1 committed:
Automate project build using Gradle

kanna-1 committed:
Add Junit Tests

kanna-1 committed:
Update README.md

kanna-1 committed:
Add minor comment

kanna-1 committed:
Merge branch 'master' of https://github.com/kanna-1/ip

kanna-1 committed:
Adjust Ui test

kanna-1 committed:
Add JavaDoc comments

kanna-1 committed:
Follow the Coding Standard

kanna-1 committed:
Add find functionality

kanna-1 committed:
Merge branch 'branch-A-JavaDoc'

kanna-1 committed:
Merge branch 'branch-A-CodingStandard'

kanna-1 committed:
Merge branch 'branch-A-JavaDoc' into Level-9

# Conflicts:
#	src/main/java/arona/task/TaskList.java
#	src/main/java/arona/ui/Ui.java

kanna-1 committed:
Merge branch 'branch-A-CodingStandard' into Level-9

kanna-1 committed:
Merge branch 'branch-A-CodingStandard'

kanna-1 committed:
Merge branch 'Level-9'

# Conflicts:
#	src/main/java/arona/task/TaskList.java
#	src/main/java/arona/ui/Ui.java

kanna-1 committed:
Merge branch 'master' of https://github.com/kanna-1/ip

kanna-1 committed:
Comply with coding standard

kanna-1 committed:
Add jar file

kanna-1 committed:
Fix jar file

kanna-1 committed:
Fix Ui test

kanna-1 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/arona/task/TaskList.java

DonovanJJ commented on src/main/java/arona/Arona.java:
```diff
@@ -0,0 +1,123 @@
+            String[] inputTokens = Parser.parseUserInput(userInput);
+            String command = Parser.getCommand(inputTokens);
+
+            if (command.equals("bye")) {
```
Perhaps you could merge these two if statements together?

DonovanJJ commented on src/main/java/arona/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,47 @@
+            return;
+        }
+        Task task = taskList.getTasks().get(taskIndex);
+        taskList.getTasks().remove(taskIndex);
```
Though I can grasp what this code is doing 👍 , I feel that it might be better if you abstract this portion out. eg. Create a new method, 
```
TaskDeleted TaskList.removeTask(taskIndex)
````
 This can help to make this part of the code neater and adhere stricter to the OOP encapsulation.

DonovanJJ commented on src/main/java/arona/commands/EventCommand.java:
```diff
@@ -0,0 +1,44 @@
+     */
+    @Override
+    public void execute() {
+        taskList.getTasks().add(eventTask);
```
Similarly here, this could be encapsulated as well

DonovanJJ commented on src/main/java/arona/commands/MarkCommand.java:
```diff
@@ -0,0 +1,46 @@
+            ui.showTaskDoesNotExist(taskIndex);
+            return;
+        }
+        taskList.getTasks().get(taskIndex).mark();
```
This could be further encapsulated within the TaskList class.

DonovanJJ commented on src/main/java/arona/Arona.java:
```diff
@@ -0,0 +1,123 @@
+                }
+            }
+            try {
+                switch (command) {
```
I like the way you encapsulate the commands within their own respective classes. 
However, I feel that this switch block should belong in the Parser class since a Parser decides what to do with the given input. In this case, Arona class seems to be the one deciding what to do based on the user input.

DonovanJJ reviewed (commented):
Overall great work, really enjoyed learning from and looking through your code 👍

kanna-1 committed:
Add comment

kanna-1 committed:
Fix merge conflict

kanna-1 committed:
Update dependencies

kanna-1 committed:
Add a GUI to the chatbot

kanna-1 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/arona/Arona.java
#	src/main/java/arona/Launcher.java

kanna-1 committed:
Add assertions for Storage.java

kanna-1 committed:
Improve code quality

kanna-1 committed:
Merge pull request #1 from kanna-1/branch-A-Assertions

Add assertions for Storage.java

kanna-1 committed:
Merge branch 'master' of https://github.com/kanna-1/ip

kanna-1 committed:
Merge pull request #2 from kanna-1/branch-A-CodeQuality

Improve code quality

kanna-1 committed:
Merge branch 'master' of https://github.com/kanna-1/ip

kanna-1 committed:
Add Undo functionality

The undo functionality in the Arona application has limited support for undoable commands, resulting in incomplete command history tracking.

To provide a comprehensive undo feature, it's essential to support all undoable commands consistently and ensure proper handling of different command types.

This commit introduces enhancements to the undo functionality:
1. The introduction of the `UndoableCommand` interface to standardize undoable commands.
2. Improved support for `MarkCommand`, `UnmarkCommand`, `DeleteCommand`, `ToDoCommand`, `EventCommand`, and `DeadlineCommand` within the undo mechanism.
3. Additional validation to check task indices to prevent out-of-range issues.

1. The `UndoableCommand` interface abstracts common undo behavior, promoting code reusability and maintainability.
2. A comprehensive conditional check for various command types ensures that all undoable actions are handled consistently.
3. Task index validation prevents errors when undoing commands related to task manipulation, improving overall robustness.

This change enhances the usability and reliability of the Arona application's undo feature, providing a more seamless user experience and preventing potential issues with command history management.

kanna-1 committed:
Merge branch 'branch-C-Undo'

kanna-1 committed:
Create and Move .jar File to Main Directory

The project's .jar file is located in the 'build' directory, causing inconvenience and confusion.

To streamline project structure and simplify execution, it's necessary to have the .jar file in the main directory.

This commit introduces the following changes:
1. Creation of the project's .jar file from the build process.
2. Relocation of the .jar file from the 'build' directory to the main directory (src).

1. Generating the .jar file during the build process ensures the most up-to-date version is included.
2. Moving the .jar file to the main directory aligns with best practices and improves project organization.

This change enhances the project's usability by providing a more straightforward way to execute the application and maintain a cleaner project structure.

Currently, the .jar file seems to have issues and may not work as expected. Further investigation is needed to address this problem.

kanna-1 committed:
Merge branch 'master' of https://github.com/kanna-1/ip

kanna-1 committed:
Update build.gradle

kanna-1 committed:
Update build.gradle

kanna-1 committed:
Remove extra file

kanna-1 committed:
Adjust Coding Style

kanna-1 committed:
Refactor code based on nus-se-bot feedback

- Rename boolean variables for clarity
- Remove arona.jar binary file
- Eliminate header comment for main method in Launcher.java

kanna-1 committed:
Update GUI

kanna-1 committed:
Refactor code based on checkstyle

kanna-1 committed:
Add User Guide

kanna-1 committed:
Refactor code to work with .jar file

kanna-1 committed:
Fix bug

kanna-1 committed:
Update User Guide

kanna-1 committed:
Update User Guide
